tests/test_database_types.py

```diff
@@ -548,8 +548,12 @@ def test_types(self, source_db, target_db, source_type, target_type, type_catego
         _insert_to_table(dst_conn, dst_table, values_in_source, target_type)
         insertion_target_duration = time.time() - start
 
-        self.table = TableSegment(self.src_conn, src_table_path, "id", None, ("col",), case_sensitive=False)
-        self.table2 = TableSegment(self.dst_conn, dst_table_path, "id", None, ("col",), case_sensitive=False)
+        if type_category == "uuid":
+            self.table = TableSegment(self.src_conn, src_table_path, "col", None, ("id",), case_sensitive=False)
+            self.table2 = TableSegment(self.dst_conn, dst_table_path, "col", None, ("id",), case_sensitive=False)
+        else:
+            self.table = TableSegment(self.src_conn, src_table_path, "id", None, ("col",), case_sensitive=False)
+            self.table2 = TableSegment(self.dst_conn, dst_table_path, "id", None, ("col",), case_sensitive=False)
```

Comment:
At first I was thinking we should change the id to be a string instead, but I'm thinking that there's going to be other weird keys we'll want to support at some point... So I'm OK leaving it like this for now.